apps/vmq_server/src/vmq_subscriber_db.erl

```diff
@@ -59,17 +59,23 @@ subscribe_db_events() ->
           when (Val == ?TOMBSTONE) or (Val == undefined) ->
             ignore;
         ({deleted, ?SUBSCRIBER_DB, SubscriberId, Subscriptions}) ->
-            Deleted = vmq_subscriber:get_changes(Subscriptions),
+            Deleted = vmq_subscriber:get_changes(get_right_subs(Subscriptions)),
             {delete, SubscriberId, Deleted};
         ({updated, ?SUBSCRIBER_DB, SubscriberId, OldVal, NewSubs})
           when (OldVal == ?TOMBSTONE) or (OldVal == undefined) ->
-            Added = vmq_subscriber:get_changes(NewSubs),
+            Added = vmq_subscriber:get_changes(get_right_subs(NewSubs)),
             {update, SubscriberId, [], Added};
         ({updated, ?SUBSCRIBER_DB, SubscriberId, OldSubs, NewSubs}) ->
-            {Removed, Added} = vmq_subscriber:get_changes(OldSubs, NewSubs),
+            {Removed, Added} = vmq_subscriber:get_changes(get_right_subs(OldSubs), get_right_subs(NewSubs)),
             {update, SubscriberId, Removed, Added};
         (_) ->
             ignore
     end.
 
+get_right_subs(Subs) when is_list(Subs) ->
+    Subs;
+get_right_subs(Subs) ->
```

Comment:
That is why i asked you why subs will be tuple and list in defferent scenario. https://github.com/erlio/vernemq/issues/241#issuecomment-265946574